mmgen/models/gans/base_translation_model.py

```diff
@@ -0,0 +1,200 @@
+        self.generators = nn.ModuleDict()
+        for style in self._reachable_styles:
+            self.generators[style] = build_module(generator)
+
+        self._disc_cfg = deepcopy(discriminator)
+        # build style-aware discriminators
+        if discriminator is not None:
+            self.discriminators = nn.ModuleDict()
+            for style in self._reachable_styles:
+                self.discriminators[style] = build_module(discriminator)
+        # support no discriminator in testing
+        else:
+            self.discriminators = None
+
+        # support no gan_loss in testing
+        if gan_loss is not None:
+            self.gan_loss = build_module(gan_loss)
+        else:
+            self.gan_loss = None
+
+        if disc_auxiliary_loss:
+            self.disc_auxiliary_losses = build_module(disc_auxiliary_loss)
+            if not isinstance(self.disc_auxiliary_losses, nn.ModuleList):
+                self.disc_auxiliary_losses = nn.ModuleList(
+                    [self.disc_auxiliary_losses])
+        else:
+            self.disc_auxiliary_loss = None
+
+        if gen_auxiliary_loss:
+            self.gen_auxiliary_losses = build_module(gen_auxiliary_loss)
+            if not isinstance(self.gen_auxiliary_losses, nn.ModuleList):
+                self.gen_auxiliary_losses = nn.ModuleList(
+                    [self.gen_auxiliary_losses])
+        else:
+            self.gen_auxiliary_losses = None
+
+        self.train_cfg = deepcopy(train_cfg) if train_cfg else None
+        self.test_cfg = deepcopy(test_cfg) if test_cfg else None
+
+        self._parse_train_cfg()
+        if test_cfg is not None:
+            self._parse_test_cfg()
+
+    def _parse_train_cfg(self):
+        """Parsing train config and set some attributes for training."""
+        if self.train_cfg is None:
+            self.train_cfg = dict()
+        # control the work flow in train step
+        self.disc_steps = self.train_cfg.get('disc_steps', 1)
+
+        self.disc_init_steps = (0 if self.train_cfg is None else
+                                self.train_cfg.get('disc_init_steps', 0))
+
+        self.real_img_key = self.train_cfg.get('real_img_key', 'real_img')
+
+        # TODO:
+
+    def _parse_test_cfg(self):
+        """Parsing test config and set some attributes for testing."""
+        if self.test_cfg is None:
+            self.test_cfg = dict()
+
+        # basic testing information
+        self.batch_size = self.test_cfg.get('batch_size', 1)
+
+    def forward(self, img, test_mode=False, **kwargs):
+        """Forward function.
+
+        Args:
+            img (tensor): Input image tensor.
+            test_mode (bool): Whether in test mode or not. Default: False.
+            kwargs (dict): Other arguments.
+        """
+        if not test_mode:
+            return self.forward_train(img, **kwargs)
+
+        return self.forward_test(img, **kwargs)
+
+    def forward_train(self, img, style, **kwargs):
```

Comment:
You may be careful to use `style`. Style transfer and image translation have obvious differences. Thus, you may consider replacing it with other words, like 'target'.